src/test/java/uk/gov/dwp/dataworks/provider/kms/KMSDataKeyGeneratorProviderTest.java

```diff
@@ -49,41 +50,41 @@ public void generateDataKey() {
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesNotFoundException() {
+    public void handlesNotFoundException() throws MasterKeystoreException {
         throwException(NotFoundException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesDisabledException() {
+    public void handlesDisabledException() throws MasterKeystoreException {
         throwException(DisabledException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesDependencyTimeoutException() {
+    public void handlesDependencyTimeoutException() throws MasterKeystoreException {
         throwException(DependencyTimeoutException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesInvalidKeyUsageException() {
+    public void handlesInvalidKeyUsageException() throws MasterKeystoreException {
         throwException(InvalidKeyUsageException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesInvalidGrantTokenException() {
+    public void handlesInvalidGrantTokenException() throws MasterKeystoreException {
         throwException(InvalidGrantTokenException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesKMSInternalException() {
+    public void handlesKMSInternalException() throws MasterKeystoreException {
         throwException(KMSInternalException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesKMSInvalidStateException() {
+    public void handlesKMSInvalidStateException() throws MasterKeystoreException {
         throwException(KMSInvalidStateException.class);
     }
```

Comment:
These test names don't tell us _why_ we are testing and _what_ they are testing. It looks just like Inheritance tree tests?

Reply:
These are the KMS tests - they have not been introduced as part of this PR - the only changes are updates to the signature to accommodate the new exception introduced. If we want to retrofit all these tests could it be done in a tech debt ticket?